### PR TITLE
miscellaneous configure fixes

### PR DESCRIPTION
**Pull request by goodell**
Small fixes for `CFLAGS` and `CPPFLAGS` handling.

@jsquyres please review at your leisure

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: configure.ac [hunk 1]
```diff
@@ -41,17 +41,17 @@ AC_ARG_ENABLE([debug],
...
 
 AC_DEFINE_UNQUOTED([ENABLE_DEBUG],[$dbg],
                    [defined to 1 if libfabric was configured with --enable-debug, 0 otherwise])
 
-dnl Fix autoconf's habit of adding -g -O2 by default
-AS_IF([test -z "$CFLAGS"],
-      [CFLAGS='-fvisibility=hidden -O2 -DNDEBUG -Wall -Wundef'])
+# Fix autoconf's habit of setting CFLAGS="-g -O2" by default while still
+# allowing the user to explicitly set CFLAGS=""
+: ${CFLAGS="-fvisibility=hidden -O2 -DNDEBUG -Wall -Wundef"}
```
Comment (jsquyres) [inline configure.ac]: I cannot find documentation for the `${VAR=VALUE}` syntax in the bash man page.  Is it functionally equivalent to `${VAR:=VALUE}` or `{VAR:-VALUE}`?

Comment (jsquyres) [inline configure.ac]: @bturrubiates points to http://www.tldp.org/LDP/abs/html/parameter-substitution.html.  Ok.

Comment (goodell) [inline configure.ac]: FWIW, my reference for this sort of thing is usually the "2.6.2 Parameter Expansion" portion of this: http://pubs.opengroup.org/onlinepubs/009604499/utilities/xcu_chap02.html

**Pull request merged**